plugins/providers/metabase/resource.go

```diff
@@ -54,6 +76,76 @@ func (d *Database) ToDomain() *domain.Resource {
 	}
 }
 
+func (t *Table) FromDomain(r *domain.Resource) error {
+	if r.Type != ResourceTypeTable {
+		return ErrInvalidResourceType
+	}
+
+	tableURN := strings.Split(r.URN, ":")
+	if len(tableURN) != 2 {
+		return ErrInvalidTableURN
+	}
+
+	tableURN = strings.Split(tableURN[1], ".")
+	id, err := strconv.Atoi(tableURN[1])
+	if err != nil {
+		return err
+	}
+
+	t.ID = id
+	t.Name = r.Name
+	t.DbId, err = strconv.Atoi(tableURN[1])
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (t *Table) ToDomain() *domain.Resource {
+	return &domain.Resource{
+		Type:    ResourceTypeTable,
+		Name:    t.Name,
+		URN:     fmt.Sprintf("table:%d.%d", t.DbId, t.ID),
+		Details: t.Database.Details,
+	}
+}
+
+func (g *Group) FromDomain(r *domain.Resource) error {
+	if r.Type != ResourceTypeGroup {
+		return ErrInvalidResourceType
+	}
+
+	groupUrn := strings.Split(r.URN, ":")
+	if len(groupUrn) != 2 {
+		return ErrInvalidDatabaseURN
+	}
+	id, err := strconv.Atoi(groupUrn[1])
+	if err != nil {
+		return err
+	}
+
+	g.ID = id
+	g.Name = r.Name
+	g.DatabaseResources = r.Details["database"].([]map[string]interface{})
```

Comment:
`map[string]interface{}` is too generic. Can this be made specific?

Reply:
done